### ISSUE TITLE
Kotlin 2.0.0 warns about Kotlin Record class annotation on "is" property

### ISSUE THREAD

**Issue by sgrimm** [jOOQ/jOOQ#16700]
### Expected behavior

Upgrading to Kotlin 2.0.0 doesn't require any changes to the jOOQ configuration.

### Actual behavior

If a table has a boolean column whose name starts with `is_`, the Kotlin generator adds a `@set:JvmName` annotation to the property in the generated Record class. As part of the fix for #12440, it also includes a suppression annotation to stop the Kotlin compiler from complaining about the setter's JVM name.

In Kotlin 2.0.0, that suppression annotation causes the compiler to output a warning:

`This code uses error suppression for 'INAPPLICABLE_JVM_NAME'. While it might compile and work, the compiler behavior is UNSPECIFIED and WON'T BE PRESERVED. Please report your use case to the Kotlin issue tracker instead: https://kotl.in/issue`

If the project is configured to treat warnings as errors, this causes a build failure.

For a pure Kotlin project, or a project where it's not important whether or not the setter follows the same naming convention as the Java generator's code, a workaround is to set `kotlinSetterJvmNameAnnotationsOnIsPrefix` to false in the generator settings, which removes both annotations and causes the Kotlin compiler to generate a setter with the default JavaBeans-style name.

As requested by the error message, I filed a Kotlin issue on the problem: https://youtrack.jetbrains.com/issue/KT-68456

### Steps to reproduce the problem

Repo with example of problem: https://github.com/sgrimm/jooq-kotlin-2-example/

CI build output showing the warning: https://github.com/sgrimm/jooq-kotlin-2-example/actions/runs/9205418662/job/25321061858

Basic repro steps demonstrated by the above repo:

1. Create a table with a boolean column whose name starts with `is_`.
2. Enable creation of Record classes in the Kotlin generator.
3. Configure the build to use Kotlin version 2.0.0.
4. Generate and build the jOOQ code.

### jOOQ Version

jOOQ community edition 3.19.8

### Database product and version

PostgreSQL 14.11 (Homebrew) on aarch64-apple-darwin23.2.0, compiled by Apple clang version 15.0.0 (clang-1500.1.0.2.5), 64-bit

### Java Version

openjdk version "21.0.3" 2024-04-16

### JDBC / R2DBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.7.3

**Comment by sgrimm** [comment]
This is probably more of a Kotlin bug than a jOOQ one; the right "fix" might be to document this behavior and maybe include a comment in the generated code to point people at the documentation (the first thing I did when I saw this warning was look at the generated code to see what it was complaining about).

**Comment by lukaseder** [comment]
Thanks a lot for the issue and the reproducer. I'll have a look at it tomorrow.

Even if it turns out to be an issue in kotlin, perhaps a workaround can be generated, it wouldn't be the first.

**Comment by lukaseder** [comment]
Ah, I see now. It's a warning promoted to error with `-Werror`. I've never turned that on in the past, because of Java's poor support for warning suppression. In my opinion, `-Werror`:

- Makes much less sense on generated code than on hand written code
- Makes sense only if *all* warnings can be suppressed. There are too many false positives in general

I'll check to see if anything can be done to prevent the warning, though.

**Comment by lukaseder** [comment]
Well, the whole topic is very messy indeed. I keep forgetting about the rabbit hole that is Kotlin's "clever" handling of `isXyz: Boolean` properties. I don't understand the original error that is being suppressed in the first place:

```
'@JvmName' annotation is not applicable to this declaration
```

It works, and I doubt it works by accident, so really curious about the feedback you're going to get on https://youtrack.jetbrains.com/issue/KT-68456

**Comment by lukaseder** [comment]
As far as I can tell, this warning goes away with Kotlin 2.1. Can you confirm? It no longer appears with your reproducer here: https://github.com/sgrimm/jooq-kotlin-2-example

Also, starting from jOOQ 3.20, we'll suppress all warnings in generated code as they tend to be mostly cosmetic or false positives:

- https://github.com/jOOQ/jOOQ/issues/17727

Closing this issue as it seems to be addressed in Kotlin 2.1.

**Comment by sgrimm** [comment]
Confirmed, this warning no longer appears in Kotlin 2.1 for me either.
